docs/source/cookbook/0_https.rst

```diff
@@ -0,0 +1,285 @@
+HTTPS
+=====
+
+This guide shows how to enable HTTPS protocol in Media Gateway. Both self-signed and signed by CA server certificates are supported. Certificates should be in PEM format.
+
+.. important::
+
+    The protocol in ``url`` field in the client configuration must be updated to ``https``.
+
+Prerequisites
+-------------
+
+* Docker
+* Docker Compose
+* openssl
+* curl
+
+Using a self-signed certificate
+-------------------------------
+
+Configuring Media Gateway
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To use a self-signed certificate update server and client configurations.
+
+.. code-block:: json
+    :caption: server
+
+    "tls": {
+        "identity": {
+            "certificate": "server.crt",
+            "key": "server.key"
+        }
+    }
+
+.. code-block:: json
+    :caption: client
+
+    "tls": {
+        "root_certificate": "server.crt"
+    }
+
+where
+
+* ``server.crt`` is a file with the server certificate in PEM format.
+
+* ``server.key`` is a file with the server key in PEM format.
+
+Generating certificates
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Generate a private key and certificate signing request
+
+.. code-block:: bash
+
+    mkdir certs
+
+    openssl genpkey -algorithm RSA -out certs/server.key
+
+    openssl req -new -key certs/server.key -out certs/server.csr -subj "/CN=media-gateway-server"
+
+If the client connects to the server by IP generate a certificate with IP subject alternative name. Otherwise generate a certificate with DNS subject alternative name.
+
+In commands below replace `192.168.0.108` and `media-gateway-server` with your values.
+
+.. code-block:: bash
+    :caption: IP SAN
+
+    export HOST_IP="192.168.0.108"
+
+    openssl x509 -req -days 365 -key certs/server.key -in certs/server.csr -out certs/server.crt -extfile <(echo "subjectAltName=IP:${HOST_IP}")
+
+.. code-block:: bash
+    :caption: DNS SAN
+
+    export MEDIA_GATEWAY_SERVER_DNS="media-gateway-server"
+
+    openssl x509 -req -days 365 -key certs/server.key -in certs/server.csr -out certs/server.crt -extfile <(echo "subjectAltName=DNS:${MEDIA_GATEWAY_SERVER_DNS}")
+
+Testing
+^^^^^^^
+
+Server
+""""""
+
+To test the server only a certificate with IP SAN is used.
+
+Prepare the configuration file
+
+.. code-block:: bash
+
+    echo "{
```

Comment:
Please replace with here docs everything with massive backslash jsons/yamls.

Reply:
What is meant by here docs?

Reply:
Fixed in https://github.com/insight-platform/MediaGateway/pull/34